### ISSUE TITLE
tuneParams/IRace fails when optimum params have NAs through requirements

### ISSUE THREAD

**Issue by mb706** [mlr-org/mlr#779]
```R
> ps = makeParamSet(makeNumericParam("C", 0.1, 0.3), makeLogicalParam("R", requires=quote(C > 1)))  # R never feasible                                                                              
> ctrl = makeTuneControlIrace(maxExperiments = 150L)                                                                                                                                                
> rdesc = makeResampleDesc("Holdout")                                                                                                                                                               
> tuneParams("classif.J48", pid.task, rdesc, par.set = ps, control = ctrl, show.info = TRUE)                                                                                                        
[Tune] Started tuning learner classif.J48 for parameter set:
     Type len Def     Constr Req Tunable Trafo
C numeric   -   - 0.1 to 0.3   -    TRUE     -
R logical   -   -          -   Y    TRUE     -
With control class: TuneControlIrace
Imputation value: 1
[Tune-x] 1: C=0.151
[Tune-y] 1: mmce.test.mean=0.281; time: 0.0 min; memory: 40Mb use, 67Mb max
[Tune-x] 2: C=0.114
[Tune-y] 2: mmce.test.mean=0.246; time: 0.0 min; memory: 40Mb use, 67Mb max
[...]
[...]
[...]
[Tune-x] 144: C=0.118
[Tune-y] 144: mmce.test.mean=0.23; time: 0.0 min; memory: 40Mb use, 67Mb max
[Tune-x] 145: C=0.108
[Tune-y] 145: mmce.test.mean=0.23; time: 0.0 min; memory: 40Mb use, 67Mb max
Error in sel.func(learner, task, resampling, measures, par.set, control,  : 
  No matching rows for final elite candidate found in opt.path! This cannot be!
```
Same for
```R
> ps = makeParamSet(makeNumericParam("C", 0.1, 0.3, requires=quote(R != R)), makeLogicalParam("R"))  # C never feasible
```
but
```R
> ps = makeParamSet(makeNumericParam("C", 0.1, 0.3), makeLogicalParam("O"))
```
works. It is because `tuneIrace` [incompletely converts from parameter values to their names as irace sees them](https://github.com/mlr-org/mlr/blob/90d40002a35ad3770d6699e48d02292fe932159a/R/tuneIrace.R#L44-L45) and gets hiccups on NAs (and probably also for discrete params that are not characters).

**Comment by mb706** [comment]
Turns out the problem happened only when a variable is never feasible during a search run (and hence the opt.path column contains only `NA`s), this happens if the number of parameters and dependencies is big enough.

**Comment by berndbischl** [comment]
fixed in PR #780
